datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -240,13 +272,15 @@ func buildCreatePayloadCases(d *schema.ResourceData) []datadogV2.SecurityMonitor
 	return payloadCases
 }
 
-func buildCreatePayloadOptions(tfOptionsList []interface{}) *datadogV2.SecurityMonitoringRuleOptions {
+func buildPayloadOptions(tfOptionsList []interface{}) *datadogV2.SecurityMonitoringRuleOptions {
 	payloadOptions := datadogV2.NewSecurityMonitoringRuleOptions()
-	var tfOptions map[string]interface{}
-	if tfOptionsList[0] == nil {
-		tfOptions = make(map[string]interface{})
-	} else {
-		tfOptions = tfOptionsList[0].(map[string]interface{})
+	tfOptions := extractMapFromInterface(tfOptionsList)
+
+	if v, ok := tfOptions["detection_method"]; ok {
+		detectionMethod := datadogV2.SecurityMonitoringRuleDetectionMethod(v.(string))
+		if detectionMethod.IsValid() {
+			payloadOptions.DetectionMethod = &detectionMethod
+		}
```

Comment:
No need for this check, it's already ensured with the ValidateFunc in the schema

Reply:
```suggestion
		payloadOptions.DetectionMethod = &detectionMethod
```